src/format.ts

```diff
@@ -110,8 +120,117 @@ function checkFormat(srcFiles: string[], baseArgs: string[]): Promise<boolean> {
     out.on('data', (data: Buffer) => {
       output += data;
     });
+
     out.on('end', () => {
-      resolve(output.indexOf('<replacement ') === -1 ? true : false);
+      findFormatErrorLines(output, options).then(() => {
+        resolve(output.indexOf('<replacement ') === -1 ? true : false);
+      });
+    });
+  });
+}
+
+/**
+ * Parses through xml string for the replacement offsets and lengths. Uses those
+ * values to locate the formatting error lines.
+ *
+ * @param output xml string
+ * @param options
+ */
+async function findFormatErrorLines(output: string, options: Options) {
+  const files = output.split('<?xml version=\'1.0\'?>\n');
+
+  for (let i = 1; i < files.length; i++) {
+    const formatErr = {
+      offset: output.match(/(?<=offset=\')(\d+)(?=\')/g),
+      length: output.match(/(?<=length=\')(\d+)(?=\')/g),
+      fix: output.match(/(?<=length=\'\d+\'>)(.*)(?=<\/replacement>)/g)
+    };
+
+    if (formatErr.length === null || formatErr.offset === null ||
+        formatErr.fix === null) {
+      return;
+    }
+
+    for (let j = 0; j < formatErr.fix.length; j++) {
+      formatErr.fix[j] = entities.decodeXML(formatErr.fix[j]);
+    }
+
+    const read = promisify(fs.readFile);
+    const argNum = 3;
+    const file = process.argv[argNum + i - 1];
+    const text = await read(file, 'utf8');
+
+    const fixed =
+        performFixes(text, formatErr.offset, formatErr.length, formatErr.fix);
+    const diff = jsdiff.structuredPatch(
+        'oldFile', 'newFile', text, fixed, 'old', 'new', {context: 3});
+    jsdiff.applyPatch('diff', diff);
+    printDiffs(file, diff.hunks, options);
+  }
+}
+
+/**
+ * Performs formatting fixes to the original string 🌷
+ *
+ * @param data original string
+ * @param errOffset
+ * @param errLength
+ * @param replacements
+ */
+function performFixes(
+    data: string, errOffset: string[], errLength: string[],
+    replacements: string[]) {
+  const replaced: string[] = [];
+  replaced.push(substring(data, 0, +errOffset[0]));
+  for (let i = 0; i < errOffset.length - 1; i++) {
+    replaced.push(replacements[i]);
+    replaced.push(
+        substring(data, +errOffset[i] + +errLength[i], +errOffset[i + 1]));
+  }
+  replaced.push(replacements[errOffset.length - 1]);
+  replaced.push(substring(
+      data, +errOffset[errOffset.length - 1] + +errLength[errOffset.length - 1],
+      Buffer.byteLength(data, 'utf8')));
+  return replaced.join('');
+}
+
+/**
+ * Prints the lines with formatting issues
+ *
+ * @param file
+ * @param diffs contains all information about the formatting changes
+ * @param options
+ */
+function printDiffs(file: string, diffs: any, options: Options) {
```

Comment:
I believe that if you `import` from `diff`, then you can get the interface that matches the `any` here.

Reply:
done